lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -121,12 +121,13 @@ IRGenDebugInfo::IRGenDebugInfo(const IRGenOptions &Opts,
   }
 
   unsigned Lang = llvm::dwarf::DW_LANG_Swift;
-  std::string Producer = version::getSwiftFullVersion();
+  std::string Producer = version::getSwiftFullVersion(
+    Opts.EffectiveLanguageVersion);
   bool IsOptimized = Opts.Optimize;
   StringRef Flags = Opts.DWARFDebugFlags;
-  unsigned Major, Minor;
-  std::tie(Major, Minor) = version::getSwiftNumericVersion();
-  unsigned MajorRuntimeVersion = Major;
+
+  assert(!Opts.EffectiveLanguageVersion.empty());
```

Comment:
As far as "what it means", it appears to thread through to an attribute on the CU DIE called `dwarf::DW_AT_APPLE_major_runtime_vers` which, as far as google can tell, nobody in the world actually uses. It showed up in 2009 in LLVM (https://github.com/apple/swift-llvm/commit/13319ceffdd99d153e62011c5dda08b95e3279e3) and it looks like possibly it was intended to be related to LLDB logic around ObjC runtime versions, but LLDB currently just sniffs the existence of a "telltale section" (https://github.com/apple/swift-lldb/blob/master/source/Plugins/LanguageRuntime/ObjC/AppleObjCRuntime/AppleObjCRuntime.cpp#L401-L409) and uses it to switch between V1 and V2.

In any case if we're emitting `3` for it now -- as we seem to be doing as of swift 3.0 -- and if ObjC doesn't have a runtime version 3, we're probably already doing wrong here. I'll add a helper function that picks a runtime version to match the current language version, explicitly.